src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaTransactionService.java

```diff
@@ -0,0 +1,57 @@
+package com.icthh.xm.tmf.ms.activation.service;
+
+import com.icthh.xm.tmf.ms.activation.domain.SagaTransaction;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SagaTransactionService {
```

Comment:
for me better to name factory or mapper or smth like this
we have SagaTransactionService it's about saga transaction logic

Reply:
I have renamed SagaTransactionService to SagaTransactionFactory.
By the way, there is no another SagaTransactionService class, SagaService contains all saga transaction logic